samples/java-eventsourced-counter-pojo/src/main/java/org/example/domain/PojoCounterProvider.java

```diff
@@ -0,0 +1,86 @@
+package org.example.domain;
+
+import com.akkaserverless.javasdk.eventsourcedentity.EventSourcedEntityContext;
+import com.akkaserverless.javasdk.eventsourcedentity.EventSourcedEntityOptions;
+import com.akkaserverless.javasdk.eventsourcedentity.EventSourcedEntityProvider;
+import com.akkaserverless.javasdk.impl.CompositeSerializer;
+import com.akkaserverless.javasdk.impl.JsonSerializer;
+import com.akkaserverless.javasdk.impl.Serializer;
+import com.example.CounterApi;
+import com.example.domain.Counter;
+import com.example.domain.CounterRouter;
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.EmptyProto;
+
+import java.util.function.Function;
+
+// This code is managed by Akka Serverless tooling.
+// It will be re-generated to reflect any changes to your protobuf definitions.
+// DO NOT EDIT
+
+/**
+ * An event sourced entity provider that defines how to register and create the entity for
+ * the Protobuf service <code>CounterService</code>.
+ * <p>
+ * Should be used with the <code>register</code> method in {@link com.akkaserverless.javasdk.AkkaServerless}.
+ */
+public class PojoCounterProvider implements EventSourcedEntityProvider<CounterState, Counter> {
+
+  private final Function<EventSourcedEntityContext, Counter> entityFactory;
+  private final EventSourcedEntityOptions options;
+
+  /**
+   * Factory method of CounterProvider
+   */
+  public static org.example.domain.PojoCounterProvider of(Function<EventSourcedEntityContext, Counter> entityFactory) {
+    return new org.example.domain.PojoCounterProvider(entityFactory, EventSourcedEntityOptions.defaults());
+  }
+
+  private PojoCounterProvider(
+      Function<EventSourcedEntityContext, Counter> entityFactory,
+      EventSourcedEntityOptions options) {
+    this.entityFactory = entityFactory;
+    this.options = options;
+  }
+
+  @Override
+  public final EventSourcedEntityOptions options() {
+    return options;
+  }
+
+  public final org.example.domain.PojoCounterProvider withOptions(EventSourcedEntityOptions options) {
+    return new org.example.domain.PojoCounterProvider(entityFactory, options);
+  }
+
+  @Override
+  public final Descriptors.ServiceDescriptor serviceDescriptor() {
+    return CounterApi.getDescriptor().findServiceByName("CounterService");
+  }
+
+  @Override
+  public final String entityType() {
+    return "counter";
+  }
+
+  @Override
+  public final CounterRouter newRouter(EventSourcedEntityContext context) {
+    return new CounterRouter(entityFactory.apply(context));
+  }
+
+
+  @Override
+  public Serializer serializer() {
+    return new CompositeSerializer(new JsonSerializer(CounterState.class))
+            .add(new JsonSerializer(ValueIncreased.class))
+            .add(new JsonSerializer(ValueDecreased.class))
+            .add(new JsonSerializer(ValueReset.class));
+  }
```

Comment:
This is what it would look like when using POJOs. 

Users could overwrite this if they want and add some extra serialization/deserfialization logic. I'm thinking on Scala developers that may choose not to use Jackson but a more idiomatic Scala json library.